File: gcc/testsuite/gfortran.dg/gomp/appendix-a/a.35.4.f90
! { dg-do compile }

      SUBROUTINE WRONG4(N)
      INTEGER N
        INTEGER I
!$OMP PARALLEL DEFAULT(SHARED)
!$OMP DO
          DO I = 1, N
             CALL WORK(I, 1)
! incorrect nesting of barrier region in a loop region
!$OMP BARRIER	! { dg-warning "may not be closely nested" }
             CALL WORK(I, 2)
          END DO
!$OMP END PARALLEL
      END SUBROUTINE WRONG4

File: gcc/testsuite/gcc.dg/gomp/appendix-a/a.35.4.c
/* { dg-do compile } */

void
wrong4 (int n)
{
#pragma omp parallel default(shared)
  {
    int i;
#pragma omp for
    for (i = 0; i < n; i++)
      {
	work (i, 0);
	/* incorrect nesting of barrier region in a loop region */
#pragma omp barrier	/* { dg-error "may not be closely nested" } */
	work (i, 1);
      }
  }
}
